### PR TITLE
Fix #3764

### LINKED ISSUE #3764: React import is being omitted
I'm getting an issue with React/JSX on VSCode with the master from today and I use the latest `react-jsx.d.ts` and `react.d.ts` from `DT`. I extend `React.Component` in an another file and then use it on my current file. So I don't expose React namespace in my current file. And when I import React it is being shaked off.

`module1.tsx` 
```typescript
/// <reference path='typings/react/react.d.ts'/>
/// <reference path='typings/react/react-jsx.d.ts'/>

import * as React from 'react'; // it didn't gave me any error when this line is omitted.

export class Document<P, S> extends React.Component<P, S> {
    public name: string;
}
```
`module2.tsx` (notice omit comment)
```typescript
/// <reference path='typings/react/react.d.ts'/>
/// <reference path='typings/react/react-jsx.d.ts'/>

import {Document} from './module1';
import * as React from 'react'; // this line is being omitted

interface Props {}
interface States {}

class Something extends Document<Props, States> {
    public name = 'helloworld';

    render() {
        return <div/>
    }
}
```
`module1.js`
```javascript
/// <reference path='typings/react/react.d.ts'/>
/// <reference path='typings/react/react-jsx.d.ts'/>
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var React = require('react');
var Document = (function (_super) {
    __extends(Document, _super);
    function Document() {
        _super.apply(this, arguments);
    }
    return Document;
})(React.Component);
exports.Document = Document;
//# sourceMappingURL=module1.js.map
```
`module2.js`
```javascript
/// <reference path='typings/react/react.d.ts'/>
/// <reference path='typings/react/react-jsx.d.ts'/>
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var module1_1 = require('./module1');
var Something = (function (_super) {
    __extends(Something, _super);
    function Something() {
        _super.apply(this, arguments);
        this.name = 'helloworld';
    }
    Something.prototype.render = function () {
        return React.createElement("div", null);
    };
    return Something;
})(module1_1.Document);
//# sourceMappingURL=module2.js.map
```
Also because the export class makes below file an external module. Don't you think it should error on below comment:
```typescript
/// <reference path='typings/react/react.d.ts'/>
/// <reference path='typings/react/react-jsx.d.ts'/>

import * as React from 'react'; // it didn't gave me any error when this line is omitted.

export class Document<P, S> extends React.Component<P, S> {
    public name: string;
}
```

### PR DESCRIPTION
**Pull request by RyanCavanaugh**
We weren't marking `React` as used when using JSX elements in `--jsx React`, which caused the import of that symbol to be elided.

Fix: Mark this symbol as used and issue an error if it can't be found.

Ref #3764

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -7299,6 +7299,16 @@ namespace ts {
             checkGrammarJsxElement(node);
             checkJsxPreconditions(node);
 
+            // If we're compiling under --jsx react, the symbol 'React' should
+            // be marked as 'used' so we don't incorrectly elide its import. And if there
+            // is no 'React' symbol in scope, we should issue an error.
+            if(compilerOptions.jsx === JsxEmit.React) {
```
Comment (DanielRosenwasser) [inline src/compiler/checker.ts]: Space after `if`

File: tests/baselines/reference/tsxElementResolution19.symbols [hunk 2]
```diff
@@ -0,0 +1,28 @@
+=== tests/cases/conformance/jsx/react.d.ts ===
+
+No type information for this code.declare module "react" {
+No type information for this code.
+No type information for this code.}
+No type information for this code.
+No type information for this code.=== tests/cases/conformance/jsx/file1.tsx ===
```
Comment (DanielRosenwasser) [inline tests/baselines/reference/tsxElementResolution19.symbols]: This is a pretty weird output.

**Comment by DanielRosenwasser** [comment]
:+1:

File: tests/baselines/reference/tsxElementResolution19.symbols (hunk 2 above)
Comment (RyanCavanaugh) [inline tests/baselines/reference/tsxElementResolution19.symbols]: Seems to be in a lot of places: https://github.com/Microsoft/TypeScript/search?utf8=%E2%9C%93&q=No+type+information+for+this+code&type=Code

**Comment by mhegazy** [comment]
:+1:

**Comment by JsonFreeman** [comment]
Wasn't this the second "marking something as used" bug having to do with JSX? This "marking something as used" thing is... pretty annoying.

**Pull request merged**